[ci] Make mac universal builds more universal

tobil4sk opened the pull request:
Right now the haxelib binary in the "universal" mac builds is only x64. To fix this we need to build haxelib with an arm64 build of neko.

First we need arm64 builds of neko, see:
https://github.com/HaxeFoundation/neko/pull/288

tobil4sk committed:
[ci] Create universal binary for haxelib

tobil4sk committed:
[ci] Install arm64 neko for arm64 mac builds

tobil4sk committed:
[ci] Build native haxelib binary on arm64 mac

tobil4sk commented:
The arm64 builds of neko are now available at https://build.haxe.org/builds/neko/mac-arm64/. However, the mac installer package script needs to be updated to use a universal binary for neko as well.

kLabz commented:
> However, the mac installer package script needs to be updated to use a universal binary for neko as well.

Hi @tobil4sk!
We'd like to move this whole universal mac thing forward to release it (on 4.3.x too), but I'm not sure what this means needs to be done exactly :thinking:

tobil4sk commented:
The next step is to create universal neko binaries. The easiest way to do this is to update the neko repo so that it also creates them.

The makefile for creating the package installer pulls the binaries from a github release, so we'll need a new neko release on github to host the binaries. We could use a prerelease to test everything is working. There's a few neko PRs that would be good to get in before a final release though.

kLabz commented:
Thanks! Though I'm not sure yet how to proceed there.

We might want to keep the neko changes to a minimum for now (4.3.5 release) though; what do you think @Simn ?

Simn commented:
I agree.

tobil4sk committed:
[make] Use universal neko binaries for mac package

It can still be configured to specific architectures if needed, using
`PACKAGE_INSTALLER_MAC_ARCH`.

tobil4sk committed:
[make] Default to host arch for mac installer

tobil4sk committed:
[make] Use neko version 2.4.0-rc

tobil4sk committed:
[make] Use neko version 2.4.0-rc.1

This version fixes a permission issue with universal mac binaries
